iceoryx_posh/source/roudi/port_manager.cpp

```diff
@@ -215,6 +219,158 @@ void PortManager::doDiscoveryForSubscriberPort(SubscriberPortType& subscriberPor
     });
 }
 
+void PortManager::destroyClientPort(popo::ClientPortRouDi::MemberType_t* const clientPortData) noexcept
+{
+    cxx::Ensures(clientPortData != nullptr && "clientPortData must not be a nullptr");
+
+    // create temporary client ports to orderly shut this client down
+    popo::ClientPortRouDi clientPortRoudi(*clientPortData);
+    popo::ClientPortUser clientPortUser(*clientPortData);
+
+    clientPortRoudi.releaseAllChunks();
+    clientPortUser.disconnect();
+
+    // process DISCONNECT for this client in RouDi and distribute it
+    clientPortRoudi.tryGetCaProMessage().and_then([this, &clientPortRoudi](auto caproMessage) {
+        cxx::Ensures(caproMessage.m_type == capro::CaproMessageType::DISCONNECT);
+
+        m_portIntrospection.reportMessage(caproMessage);
+        this->sendToAllMatchingServerPorts(caproMessage, clientPortRoudi);
+    });
+
+    /// @todo iox-#27 report to port introspection
+
+    // delete client port from list after DISCONNECT was processed
+    m_portPool->removeClientPort(clientPortData);
+
+    LogDebug() << "Destroyed client port";
+}
+
+void PortManager::handleClientPorts() noexcept
+{
+    // get requests for change of connection state of clients
+    for (auto clientPortData : m_portPool->getClientPortDataList())
+    {
+        popo::ClientPortRouDi clientPort(*clientPortData);
+
+        doDiscoveryForClientPort(clientPort);
+
+        // check if we have to destroy this clinet port
+        if (clientPort.toBeDestroyed())
+        {
+            destroyClientPort(clientPortData);
+        }
+    }
+}
+
+void PortManager::doDiscoveryForClientPort(popo::ClientPortRouDi& clientPort) noexcept
+{
+    clientPort.tryGetCaProMessage().and_then([this, &clientPort](auto caproMessage) {
+        if ((capro::CaproMessageType::CONNECT == caproMessage.m_type)
+            || (capro::CaproMessageType::DISCONNECT == caproMessage.m_type))
+        {
+            /// @todo iox-#27 report to port introspection
+            if (!this->sendToAllMatchingServerPorts(caproMessage, clientPort))
+            {
+                LogDebug() << "capro::CONNECT/DISCONNECT, no matching server!!";
```

Comment:
Could you add the process name of the port owner and the service description? This goes also for all the other log messages.

Reply:
same as above, the process name is not available ... or did you mean the runtime name?